hi-ml/src/health_ml/run_ml.py

```diff
@@ -274,81 +268,78 @@ def is_crossval_disabled_or_child_0(self) -> bool:
+        if self.container.run_extra_val_epoch:
+            self.container.on_run_extra_validation_epoch()
+        self.set_trainer_for_inference()
+        self.data_module = self.container.get_data_module()
 
     def run_training(self) -> None:
         """
         The main training loop. It creates the Pytorch model based on the configuration options passed in,
         creates a Pytorch Lightning trainer, and trains the model.
         If a checkpoint was specified, then it loads the checkpoint before resuming training.
+        The cwd is changed to the outputs folder so that the model can write to current working directory, and still
+        everything is put into the right place in AzureML (only the contents of the "outputs" folder is treated as a
+        result file).
         """
-        # Change to the outputs folder so that the model can write to current working directory, and still everything
-        # is put into the right place in AzureML (only the contents of the "outputs" folder is treated as a result file)
         with change_working_directory(self.container.outputs_folder):
             assert self.trainer, "Trainer should be initialized before training. Call self.init_training() first."
-            self.trainer.fit(self.container.model, datamodule=self.container.get_data_module())
-
+            self.trainer.fit(self.container.model, datamodule=self.data_module)
         for logger in self.trainer.loggers:
             assert logger is not None
             logger.finalize('success')
 
     def run_validation(self) -> None:
-        """
-        Run validation on the validation set for all models to save time/memory consuming outputs.
-        """
-        self.container.on_run_extra_validation_epoch()
-        trainer = self.get_trainer_for_inference(checkpoint_path=None)
-        with change_working_directory(self.container.outputs_folder):
-            trainer.validate(self.container.model, datamodule=self.container.get_data_module())
-
-    def validate_model_weights(self) -> None:
-        logging.info("Validating model weights.")
-        weights = torch.load(self.checkpoint_handler.get_checkpoint_to_test())["state_dict"]
-        number_mismatch = 0
-        for name, param in self.container.model.named_parameters():
-            if not torch.allclose(weights[name].cpu(), param):
-                logging.warning(f"Parameter {name} does not match between model and checkpoint.")
-                number_mismatch += 1
-        logging.info(f"Number of mismatched parameters: {number_mismatch}")
+        """Run validation on the validation set for all models to save time/memory consuming outputs. This is done in
+        inference only mode or when the user has requested an extra validation epoch. The cwd is changed to the outputs
+        folder """
+        if self.container.run_extra_val_epoch or self.container.run_inference_only:
```

Comment:
If run_inference_only is set, why do we want to run validation?

Reply:
Maybe we want to have a commandline flag there?

Reply:
see comment above about init_inference 
I think we should rename inference to evaluation
Also instead of adding an extra flag what about using a `run_mode`
where, 
RunMode: 
- all: training, validation, testing
- validation 
- testing
- evaluation: both validation and testing?

Reply:
Will address this in a separate PR